File: app/models/subdeal.rb
class Subdeal
  include DataMapper::Resource
  
  property :id, Serial
  property :title, String, :required => true, :default => ""
  property :coupon_title, String, :required => true, :default => ""
  property :regular_price, Decimal, :scale => 2, :required => true
  property :discount_price, Decimal, :scale => 2, :required => true 
  property :created_ts, DateTime, :default => ::Constant::MIN_TIME
  property :update_ts, DateTime, :default => ::Constant::MIN_TIME
  property :deleted_ts, ParanoidDateTime
  #property :deleted, ParanoidBoolean, :default => false
  
  belongs_to :deal  
end

File: public/javascripts/mobile/app/model/SubDeal.js
Ext.define('Genesis.model.SubDeal',
{
   extend : 'Ext.data.Model',
   id : 'SubDeal',
   config :
   {
      belongsTo : 'Genesis.model.Deal',
      fields : ['title', 'coupon_title', 'regular_price', 'discount_price', 'created_ts', 'update_ts']
   }
});
